staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/metrics.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidc
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"hash"
+	"k8s.io/utils/clock"
+	"sync"
+	"time"
+
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "authentication"
+)
+
+var (
+	jwtAuthenticatorLatencyMetric = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "jwt_authenticator_latency_seconds",
+			Help:           "Latency of jwt authentication operations in seconds. This is the time spent authenticating a token for cache miss only (i.e. when the token is not found in the cache).",
+			StabilityLevel: metrics.ALPHA,
+			// Start 0.1ms with the last bucket being [~52s, +Inf)
+			Buckets: metrics.ExponentialBuckets(0.0001, 2, 20),
+		},
+		[]string{"result", "jwt_issuer_hash", "apiserver_id_hash"},
```

Comment:
remind me why we partition some metrics by apiserver_id_hash and not others?

I thought the apiserver_id_hash partitioning was around operations that were server-specific like reloading config files (authz and encryption config files)

Reply:
> I thought the apiserver_id_hash partitioning was around operations that were server-specific like reloading config files (authz and encryption config files)

This is correct. The reason I added for this metric is because the config reload could have happened in a single API server and the user can see which one as part of the latency metric. If we prefer to just have the `apiserver_id_hash` in the config reload metrics, I think it's fine to drop it here.

Reply:
yeah, I think that's more coherent with what we do for authz and kms

Reply:
removed `apiserver_id_hash` from the metric.